### PR TITLE
[Merged by Bors] - Split time functionality into bevy_time

### PR DESCRIPTION
**Pull request by CAD97**
# Objective

Reduce the catch-all grab-bag of functionality in bevy_core by minimally splitting off time functionality into bevy_time. Functionality like that provided by #3002 would increase the complexity of bevy_time, so this is a good candidate for pulling into its own unit.

A step in addressing #2931 and splitting bevy_core into more specific locations.

## Solution

Pull the time module of bevy_core into a new crate, bevy_time.

# Migration guide

- Time related types (e.g. `Time`, `Timer`, `Stopwatch`, `FixedTimestep`, etc.) should be imported from `bevy::time::*` rather than `bevy::core::*`.
- If you were adding `CorePlugin` manually, you'll also want to add `TimePlugin` from `bevy::time`.
- The `bevy::core::CorePlugin::Time` system label is replaced with `bevy::time::TimeSystem`.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
I definitely like reducing the chaos of our crats further!

> CorePlugin still handles registering bevy_time onto the App. Pulling out a TimePlugin is likely the next step at reducing bevy_core's catchall functionality, if we want to continue down this way.

I think I like that: it would be nice to be able to swap it out or remove it for niche use cases.

**Comment by mockersf** [comment]
as it is, this would increase build time as it split a new crate without increasing parallelism

**Comment by CAD97** [comment]
As it turns out, bevy_diagnostic is the only in-tree user of "bevy_time", and only uses bevy_time, not the rest of the bevy_core functionality. As such, I've gone ahead and made it depend on bevy_time directly; this also means that the compilation dependency graph is reordered and no longer _strictly_ just a pessimization.

I can pull out a `TimePlugin` in this PR, but doing enough to make bevy_core no longer use bevy_time seems a bit much for one PR? Obviously that'd be the follow-up.

**Comment by alice-i-cecile** [comment]
>I can pull out a TimePlugin in this PR, but doing enough to make bevy_core no longer use bevy_time seems a bit much for one PR? Obviously that'd be the follow-up.

Alright, sounds good to me.

**Comment by CAD97** [comment]
Things continue to be easier to do than I expect (and that's a good thing! Good job everyone!) so I just went all the way to separate bevy_time from bevy_core. I am honestly surprised at how little actually needed to change to pull TimePlugin out of CorePlugin.

File: crates/bevy_time/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+impl Plugin for TimePlugin {
+    fn build(&self, app: &mut App) {
+        app.init_resource::<Time>()
+            .init_resource::<FixedTimesteps>()
+            .register_type::<Timer>()
+            // time system is added as an "exclusive system" to ensure it runs before other systems
+            // in CoreStage::First
+            .add_system_to_stage(
+                CoreStage::First,
+                time_system.exclusive_system().label(TimeSystem),
```
Comment (alice-i-cecile) [inline crates/bevy_time/src/lib.rs:39]:
```suggestion
                time_system.exclusive_system().at_start().label(TimeSystem),
```

This fixes the timing: by default exclusive systems run at the end of the stage, not the beginning.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Small nit about system timing: I think this fixes an existing bug, but I'd rather not let it continue now that I've seen it.

Other than that, this LGTM: the port went very smoothly, I don't see any mistakes and I'm pretty firmly in favor of the general direction of splitting out `bevy_core` into more useful parts.

Cart, if and when you merge this, please add a `A - Time` label.

File: crates/bevy_time/src/lib.rs (hunk 1 above)
Comment (CAD97) [reply crates/bevy_time/src/lib.rs:39]: Did this change since bevy_core was first written? Because this is just verbatim what was there previously.

**Comment by CAD97** [comment]
Only CI error seems unrelated

```
error[B004]: found 2 duplicate entries for crate 'wasi'
    ┌─ /home/runner/work/bevy/bevy/Cargo.lock:296:1
    │  
296 │ ╭ wasi 0.10.2+wasi-snapshot-preview1 registry+https://github.com/rust-lang/crates.io-index
297 │ │ wasi 0.11.0+wasi-snapshot-preview1 registry+https://github.com/rust-lang/crates.io-index
    │ ╰────────────────────────────────────────────────────────────────────────────────────────^ lock entries
```

<details><summary>tree</summary>

```
    = wasi v0.10.2+wasi-snapshot-preview1
      └── getrandom v0.2.5
          ├── ahash v0.7.6
          │   ├── bevy_utils v0.6.0
          │   │   ├── bevy_app v0.6.0
          │   │   │   ├── bevy_asset v0.6.0
          │   │   │   │   ├── bevy_audio v0.6.0
          │   │   │   │   │   └── bevy_internal v0.6.0
          │   │   │   │   │       ├── bevy v0.6.0
          │   │   │   │   │       ├── bevy v0.6.0 (*)
          │   │   │   │   │       ├── (dev) bevy_audio v0.6.0 (*)
          │   │   │   │   │       └── (dev) bevy_log v0.6.0
          │   │   │   │   │           ├── (dev) bevy_app v0.6.0 (*)
          │   │   │   │   │           ├── bevy_asset v0.6.0 (*)
          │   │   │   │   │           ├── bevy_diagnostic v0.6.0
          │   │   │   │   │           │   ├── bevy_asset v0.6.0 (*)
          │   │   │   │   │           │   └── bevy_internal v0.6.0 (*)
          │   │   │   │   │           ├── bevy_gltf v0.6.0
          │   │   │   │   │           │   └── bevy_internal v0.6.0 (*)
          │   │   │   │   │           ├── bevy_internal v0.6.0 (*)
          │   │   │   │   │           ├── bevy_sprite v0.6.0
          │   │   │   │   │           │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   │           │   ├── bevy_text v0.6.0
bans FAILED
          │   │   │   │   │           │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   │           │   │   └── bevy_ui v0.6.0
          │   │   │   │   │           │   │       └── bevy_internal v0.6.0 (*)
          │   │   │   │   │           │   └── bevy_ui v0.6.0 (*)
          │   │   │   │   │           └── bevy_ui v0.6.0 (*)
          │   │   │   │   ├── bevy_core_pipeline v0.6.0
          │   │   │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   │   ├── bevy_pbr v0.6.0
          │   │   │   │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   │   │   │   └── bevy_internal v0.6.0 (*)
          │   │   │   │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   │   │   │   └── bevy_ui v0.6.0 (*)
          │   │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   │   │   ├── bevy_render v0.6.0
          │   │   │   │   │   ├── bevy_core_pipeline v0.6.0 (*)
          │   │   │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   │   │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   │   │   │   ├── bevy_text v0.6.0 (*)
          │   │   │   │   │   └── bevy_ui v0.6.0 (*)
          │   │   │   │   ├── bevy_scene v0.6.0
          │   │   │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   │   │   └── bevy_internal v0.6.0 (*)
          │   │   │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   │   │   ├── bevy_text v0.6.0 (*)
          │   │   │   │   └── bevy_ui v0.6.0 (*)
          │   │   │   ├── bevy_audio v0.6.0 (*)
          │   │   │   ├── bevy_core v0.6.0
          │   │   │   │   ├── (dev) bevy_asset v0.6.0 (*)
          │   │   │   │   ├── bevy_core_pipeline v0.6.0 (*)
          │   │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   │   │   ├── bevy_render v0.6.0 (*)
          │   │   │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   │   │   ├── bevy_text v0.6.0 (*)
          │   │   │   │   └── bevy_ui v0.6.0 (*)
          │   │   │   ├── bevy_core_pipeline v0.6.0 (*)
          │   │   │   ├── bevy_diagnostic v0.6.0 (*)
          │   │   │   ├── bevy_gilrs v0.6.0
          │   │   │   │   └── bevy_internal v0.6.0 (*)
          │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   ├── bevy_input v0.6.0
          │   │   │   │   ├── bevy_gilrs v0.6.0 (*)
          │   │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   ├── bevy_ui v0.6.0 (*)
          │   │   │   │   └── bevy_winit v0.6.0
          │   │   │   │       └── bevy_internal v0.6.0 (*)
          │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   ├── bevy_log v0.6.0 (*)
          │   │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   │   ├── bevy_render v0.6.0 (*)
          │   │   │   ├── bevy_scene v0.6.0 (*)
          │   │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   │   ├── bevy_text v0.6.0 (*)
          │   │   │   ├── bevy_time v0.6.0
          │   │   │   │   ├── bevy_diagnostic v0.6.0 (*)
          │   │   │   │   └── bevy_internal v0.6.0 (*)
          │   │   │   ├── bevy_transform v0.6.0
          │   │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   │   │   ├── bevy_render v0.6.0 (*)
          │   │   │   │   ├── bevy_scene v0.6.0 (*)
          │   │   │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   │   │   ├── bevy_text v0.6.0 (*)
          │   │   │   │   └── bevy_ui v0.6.0 (*)
          │   │   │   ├── bevy_ui v0.6.0 (*)
          │   │   │   ├── bevy_window v0.6.0
          │   │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   │   │   ├── bevy_render v0.6.0 (*)
          │   │   │   │   ├── bevy_text v0.6.0 (*)
          │   │   │   │   ├── bevy_ui v0.6.0 (*)
          │   │   │   │   └── bevy_winit v0.6.0 (*)
          │   │   │   └── bevy_winit v0.6.0 (*)
          │   │   ├── bevy_asset v0.6.0 (*)
          │   │   ├── bevy_audio v0.6.0 (*)
          │   │   ├── bevy_core v0.6.0 (*)
          │   │   ├── bevy_core_pipeline v0.6.0 (*)
          │   │   ├── bevy_diagnostic v0.6.0 (*)
          │   │   ├── bevy_ecs v0.6.0
          │   │   │   ├── bevy_app v0.6.0 (*)
          │   │   │   ├── bevy_asset v0.6.0 (*)
          │   │   │   ├── bevy_audio v0.6.0 (*)
          │   │   │   ├── bevy_core v0.6.0 (*)
          │   │   │   ├── bevy_core_pipeline v0.6.0 (*)
          │   │   │   ├── bevy_diagnostic v0.6.0 (*)
          │   │   │   ├── bevy_gilrs v0.6.0 (*)
          │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   ├── bevy_input v0.6.0 (*)
          │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   │   ├── bevy_render v0.6.0 (*)
          │   │   │   ├── bevy_scene v0.6.0 (*)
          │   │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   │   ├── bevy_text v0.6.0 (*)
          │   │   │   ├── bevy_time v0.6.0 (*)
          │   │   │   ├── bevy_transform v0.6.0 (*)
          │   │   │   ├── bevy_ui v0.6.0 (*)
          │   │   │   ├── bevy_window v0.6.0 (*)
          │   │   │   └── bevy_winit v0.6.0 (*)
          │   │   ├── bevy_gilrs v0.6.0 (*)
          │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   ├── bevy_input v0.6.0 (*)
          │   │   ├── bevy_internal v0.6.0 (*)
          │   │   ├── bevy_log v0.6.0 (*)
          │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   ├── bevy_reflect v0.6.0
          │   │   │   ├── bevy_app v0.6.0 (*)
          │   │   │   ├── bevy_asset v0.6.0 (*)
          │   │   │   ├── bevy_audio v0.6.0 (*)
          │   │   │   ├── bevy_core v0.6.0 (*)
          │   │   │   ├── bevy_ecs v0.6.0 (*)
          │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   ├── bevy_math v0.6.0
          │   │   │   │   ├── bevy_core v0.6.0 (*)
          │   │   │   │   ├── bevy_gltf v0.6.0 (*)
          │   │   │   │   ├── bevy_input v0.6.0 (*)
          │   │   │   │   ├── bevy_internal v0.6.0 (*)
          │   │   │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   │   │   ├── bevy_render v0.6.0 (*)
          │   │   │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   │   │   ├── bevy_text v0.6.0 (*)
          │   │   │   │   ├── bevy_transform v0.6.0 (*)
          │   │   │   │   ├── bevy_ui v0.6.0 (*)
          │   │   │   │   ├── bevy_window v0.6.0 (*)
          │   │   │   │   └── bevy_winit v0.6.0 (*)
          │   │   │   ├── bevy_pbr v0.6.0 (*)
          │   │   │   ├── bevy_render v0.6.0 (*)
          │   │   │   ├── bevy_scene v0.6.0 (*)
          │   │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   │   ├── bevy_text v0.6.0 (*)
          │   │   │   ├── bevy_time v0.6.0 (*)
          │   │   │   ├── bevy_transform v0.6.0 (*)
          │   │   │   └── bevy_ui v0.6.0 (*)
          │   │   ├── bevy_render v0.6.0 (*)
          │   │   ├── bevy_scene v0.6.0 (*)
          │   │   ├── bevy_sprite v0.6.0 (*)
          │   │   ├── bevy_text v0.6.0 (*)
          │   │   ├── bevy_time v0.6.0 (*)
          │   │   ├── bevy_transform v0.6.0 (*)
          │   │   ├── bevy_ui v0.6.0 (*)
          │   │   ├── bevy_window v0.6.0 (*)
          │   │   └── bevy_winit v0.6.0 (*)
          │   └── hashbrown v0.11.2
          │       ├── bevy_utils v0.6.0 (*)
          │       ├── gpu-descriptor v0.2.2
          │       │   └── wgpu-hal v0.12.4
          │       │       ├── wgpu v0.12.0
          │       │       │   └── bevy_render v0.6.0 (*)
          │       │       ├── wgpu-core v0.12.2
          │       │       │   ├── wgpu v0.12.0 (*)
          │       │       │   └── wgpu v0.12.0 (*)
          │       │       ├── wgpu-core v0.12.2 (*)
          │       │       ├── wgpu-core v0.12.2 (*)
          │       │       ├── wgpu-core v0.12.2 (*)
          │       │       └── wgpu-core v0.12.2 (*)
          │       └── indexmap v1.8.0
          │           ├── naga v0.8.5
          │           │   ├── bevy_render v0.6.0 (*)
          │           │   ├── wgpu v0.12.0 (*)
          │           │   ├── wgpu v0.12.0 (*)
          │           │   ├── wgpu-core v0.12.2 (*)
          │           │   └── wgpu-hal v0.12.4 (*)
          │           ├── petgraph v0.6.0
          │           │   └── naga v0.8.5 (*)
          │           ├── serde_yaml v0.8.23
          │           │   └── insta v0.16.1
          │           │       └── (dev) bevy_crevice v0.6.0
          │           │           └── bevy_render v0.6.0 (*)
          │           └── toml v0.5.8
          │               ├── cargo-manifest v0.2.6
          │               │   └── bevy_macro_utils v0.6.0
          │               │       ├── bevy-crevice-derive v0.6.0
          │               │       │   └── bevy_crevice v0.6.0 (*)
          │               │       ├── bevy_derive v0.6.0
          │               │       │   ├── bevy_app v0.6.0 (*)
          │               │       │   ├── bevy_core v0.6.0 (*)
          │               │       │   ├── bevy_internal v0.6.0 (*)
          │               │       │   ├── bevy_render v0.6.0 (*)
          │               │       │   ├── bevy_ui v0.6.0 (*)
          │               │       │   └── bevy_utils v0.6.0 (*)
          │               │       ├── bevy_ecs_macros v0.6.0
          │               │       │   └── bevy_ecs v0.6.0 (*)
          │               │       └── bevy_reflect_derive v0.6.0
          │               │           └── bevy_reflect v0.6.0 (*)
          │               └── proc-macro-crate v1.1.3
          │                   ├── ndk-macro v0.3.0
          │                   │   ├── ndk-glue v0.5.1
          │                   │   │   ├── bevy_asset v0.6.0 (*)
          │                   │   │   ├── bevy_internal v0.6.0 (*)
          │                   │   │   └── winit v0.26.1
          │                   │   │       ├── bevy_winit v0.6.0 (*)
          │                   │   │       └── bevy_winit v0.6.0 (*)
          │                   │   └── ndk-glue v0.6.1
          │                   │       ├── cpal v0.13.5
          │                   │       │   └── rodio v0.15.0
          │                   │       │       ├── bevy_audio v0.6.0 (*)
          │                   │       │       └── bevy_audio v0.6.0 (*)
          │                   │       └── oboe v0.4.5
          │                   │           └── cpal v0.13.5 (*)
          │                   └── num_enum_derive v0.5.7
          │                       └── num_enum v0.5.7
          │                           ├── ndk v0.5.0
          │                           │   ├── ndk-glue v0.5.1 (*)
          │                           │   └── winit v0.26.1 (*)
          │                           └── ndk v0.6.0
          │                               ├── cpal v0.13.5 (*)
          │                               ├── ndk-glue v0.6.1 (*)
          │                               └── oboe v0.4.5 (*)
          ├── bevy_utils v0.6.0 (*)
          ├── rand_core v0.6.3
          │   ├── rand v0.8.5
          │   │   ├── (dev) bevy v0.6.0 (*)
          │   │   ├── bevy_asset v0.6.0 (*)
          │   │   └── (dev) bevy_ecs v0.6.0 (*)
          │   └── rand_chacha v0.3.1
          │       └── rand v0.8.5 (*)
          └── uuid v0.8.2
              ├── bevy_reflect_derive v0.6.0 (*)
              ├── bevy_scene v0.6.0 (*)
              ├── bevy_utils v0.6.0 (*)
              ├── bevy_utils v0.6.0 (*)
              ├── gilrs v0.8.2
              │   └── bevy_gilrs v0.6.0 (*)
              └── gilrs-core v0.3.2
                  └── gilrs v0.8.2 (*)
    = wasi v0.11.0+wasi-snapshot-preview1
      └── mio v0.8.1
          └── winit v0.26.1
              ├── bevy_winit v0.6.0
              │   └── bevy_internal v0.6.0
              │       ├── bevy v0.6.0
              │       ├── bevy v0.6.0 (*)
              │       ├── (dev) bevy_audio v0.6.0
              │       │   └── bevy_internal v0.6.0 (*)
              │       └── (dev) bevy_log v0.6.0
              │           ├── (dev) bevy_app v0.6.0
              │           │   ├── bevy_asset v0.6.0
              │           │   │   ├── bevy_audio v0.6.0 (*)
              │           │   │   ├── bevy_core_pipeline v0.6.0
              │           │   │   │   ├── bevy_internal v0.6.0 (*)
              │           │   │   │   ├── bevy_pbr v0.6.0
              │           │   │   │   │   ├── bevy_gltf v0.6.0
              │           │   │   │   │   │   └── bevy_internal v0.6.0 (*)
              │           │   │   │   │   └── bevy_internal v0.6.0 (*)
              │           │   │   │   ├── bevy_sprite v0.6.0
              │           │   │   │   │   ├── bevy_internal v0.6.0 (*)
              │           │   │   │   │   ├── bevy_text v0.6.0
              │           │   │   │   │   │   ├── bevy_internal v0.6.0 (*)
              │           │   │   │   │   │   └── bevy_ui v0.6.0
              │           │   │   │   │   │       └── bevy_internal v0.6.0 (*)
              │           │   │   │   │   └── bevy_ui v0.6.0 (*)
              │           │   │   │   └── bevy_ui v0.6.0 (*)
              │           │   │   ├── bevy_gltf v0.6.0 (*)
              │           │   │   ├── bevy_internal v0.6.0 (*)
              │           │   │   ├── bevy_pbr v0.6.0 (*)
              │           │   │   ├── bevy_render v0.6.0
              │           │   │   │   ├── bevy_core_pipeline v0.6.0 (*)
              │           │   │   │   ├── bevy_gltf v0.6.0 (*)
              │           │   │   │   ├── bevy_internal v0.6.0 (*)
              │           │   │   │   ├── bevy_pbr v0.6.0 (*)
              │           │   │   │   ├── bevy_sprite v0.6.0 (*)
              │           │   │   │   ├── bevy_text v0.6.0 (*)
              │           │   │   │   └── bevy_ui v0.6.0 (*)
              │           │   │   ├── bevy_scene v0.6.0
              │           │   │   │   ├── bevy_gltf v0.6.0 (*)
              │           │   │   │   └── bevy_internal v0.6.0 (*)
              │           │   │   ├── bevy_sprite v0.6.0 (*)
              │           │   │   ├── bevy_text v0.6.0 (*)
              │           │   │   └── bevy_ui v0.6.0 (*)
              │           │   ├── bevy_audio v0.6.0 (*)
              │           │   ├── bevy_core v0.6.0
              │           │   │   ├── (dev) bevy_asset v0.6.0 (*)
              │           │   │   ├── bevy_core_pipeline v0.6.0 (*)
              │           │   │   ├── bevy_gltf v0.6.0 (*)
              │           │   │   ├── bevy_internal v0.6.0 (*)
              │           │   │   ├── bevy_pbr v0.6.0 (*)
              │           │   │   ├── bevy_render v0.6.0 (*)
              │           │   │   ├── bevy_sprite v0.6.0 (*)
              │           │   │   ├── bevy_text v0.6.0 (*)
              │           │   │   └── bevy_ui v0.6.0 (*)
              │           │   ├── bevy_core_pipeline v0.6.0 (*)
              │           │   ├── bevy_diagnostic v0.6.0
              │           │   │   ├── bevy_asset v0.6.0 (*)
              │           │   │   └── bevy_internal v0.6.0 (*)
              │           │   ├── bevy_gilrs v0.6.0
              │           │   │   └── bevy_internal v0.6.0 (*)
              │           │   ├── bevy_gltf v0.6.0 (*)
              │           │   ├── bevy_input v0.6.0
              │           │   │   ├── bevy_gilrs v0.6.0 (*)
              │           │   │   ├── bevy_internal v0.6.0 (*)
              │           │   │   ├── bevy_ui v0.6.0 (*)
              │           │   │   └── bevy_winit v0.6.0 (*)
              │           │   ├── bevy_internal v0.6.0 (*)
              │           │   ├── bevy_log v0.6.0 (*)
              │           │   ├── bevy_pbr v0.6.0 (*)
              │           │   ├── bevy_render v0.6.0 (*)
              │           │   ├── bevy_scene v0.6.0 (*)
              │           │   ├── bevy_sprite v0.6.0 (*)
              │           │   ├── bevy_text v0.6.0 (*)
              │           │   ├── bevy_time v0.6.0
              │           │   │   ├── bevy_diagnostic v0.6.0 (*)
              │           │   │   └── bevy_internal v0.6.0 (*)
              │           │   ├── bevy_transform v0.6.0
              │           │   │   ├── bevy_gltf v0.6.0 (*)
              │           │   │   ├── bevy_internal v0.6.0 (*)
              │           │   │   ├── bevy_pbr v0.6.0 (*)
              │           │   │   ├── bevy_render v0.6.0 (*)
              │           │   │   ├── bevy_scene v0.6.0 (*)
              │           │   │   ├── bevy_sprite v0.6.0 (*)
              │           │   │   ├── bevy_text v0.6.0 (*)
              │           │   │   └── bevy_ui v0.6.0 (*)
              │           │   ├── bevy_ui v0.6.0 (*)
              │           │   ├── bevy_window v0.6.0
              │           │   │   ├── bevy_internal v0.6.0 (*)
              │           │   │   ├── bevy_pbr v0.6.0 (*)
              │           │   │   ├── bevy_render v0.6.0 (*)
              │           │   │   ├── bevy_text v0.6.0 (*)
              │           │   │   ├── bevy_ui v0.6.0 (*)
              │           │   │   └── bevy_winit v0.6.0 (*)
              │           │   └── bevy_winit v0.6.0 (*)
              │           ├── bevy_asset v0.6.0 (*)
              │           ├── bevy_diagnostic v0.6.0 (*)
              │           ├── bevy_gltf v0.6.0 (*)
              │           ├── bevy_internal v0.6.0 (*)
              │           ├── bevy_sprite v0.6.0 (*)
              │           └── bevy_ui v0.6.0 (*)
              └── bevy_winit v0.6.0 (*)
```

**Comment by alice-i-cecile** [comment]
> Only CI error seems unrelated

Yep, not your fault.

File: crates/bevy_time/src/lib.rs (hunk 1 above)
Comment (maniwani) [reply crates/bevy_time/src/lib.rs]: > by default exclusive systems run at the end of the stage, not the beginning

This is not correct. `AtStart` is the [default insertion point][1], so adding the specifier isn't necessary.

[1]: https://github.com/bevyengine/bevy/blob/0e821da7045852ff677c2b8ee945dc85a899b181/crates/bevy_ecs/src/schedule/system_descriptor.rs#L241-L251

**Comment by alice-i-cecile** [comment]
bors try

**Comment by CAD97** [comment]
Rebased.

This will almost certainly fail but might as well try:

bors try=@alice-i-cecile

**Review by mockersf — APPROVED** [review]
not sure I agree this is needed, but it's well done

**Review by IceSentry — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
bors try

**Comment by alice-i-cecile** [comment]
@CAD97 can you rebase? I'd like to get this in, in order to make future PRs easier.

**Comment by CAD97** [comment]
@alice-i-cecile rebased!

**Comment by CAD97** [comment]
(gah, I missed an unused dep in the rebase. I'll get to it tomorrow when I'm back at desktop.)

**Comment by CAD97** [comment]
Fixed the udep.

**Comment by CAD97** [comment]
Rebased for #4469 and removed an accidental unused `bevy_time/src/mod.rs` that appeared in a previous rebase.

**Comment by CAD97** [comment]
(Meta note: without a ping when a merge conflict arises, which GitHub does not provide by default, noticing that there's a merge conflict in order to rebase relies on the PR author happening to revisit the PR to see the notice, or a reviewer giving the ping.)

**Comment by CAD97** [comment]
"dead link" on this rebase was a 503 response from <https://opensource.org/licenses/MIT>

**Comment by james7132** [comment]
bors try

**Review by cart — APPROVED** [review]
Just added `bevy_time` to the publish script. I think this is good to go.

**Pull request closed without merging**